api/proto/teleport/integration/v1/awsoidc_service.proto

```diff
@@ -32,6 +36,48 @@ service AWSOIDCService {
   rpc DeployDatabaseService(DeployDatabaseServiceRequest) returns (DeployDatabaseServiceResponse);
 }
 
+// ListEICERequest is a request for a paginated list of AWS EC2 Instance Connect Endpoints.
+message ListEICERequest {
+  // Integration is the AWS OIDC Integration name.
+  string integration = 1;
+  // Region is the AWS Region
+  string region = 2;
+  // VPCIDs is used to filter for EICEs of those VPCs.
+  repeated string vpc_ids = 3;
+  // NextToken is the token to be used to fetch the next page.
+  // If empty, the first page is fetched.
+  string next_token = 4;
+}
+
+// EC2InstanceConnectEndpoint is a representation of a Amazon VPC EC2 Instance Connect Endpoint.
+message EC2InstanceConnectEndpoint {
+  // Name is the endpoint name.
+  string name = 1;
+  // State is the endpoint state.
+  // Known values:
+  // create-in-progress | create-complete | create-failed | delete-in-progress | delete-complete | delete-failed
+  string state = 2;
+  // StateMessage contains a message describing the sate of the EICE.
+  string state_message = 3;
+  // DashboardLink is a URL to AWS Console where the user can see the EC2 Instance Connect Endpoint.
+  string dashboard_link = 4;
+  // SubnetID is the subnet used by the endpoint.
+  string subnet_id = 5;
+  // VPCID is the VPC ID where the Endpoint is created.
+  string vpc_id = 6;
+}
+
+// ListEICEResponse contains a page of AWS EC2 Instance Connect Endpoints.
+message ListEICEResponse {
+  // EC2ICEs contains the page of EC2 Instance Connect Endpoints.
+  repeated EC2InstanceConnectEndpoint ec2ices = 1;
+  // DashboardLink is the URL for AWS Web Console that lists all the Endpoints for the queries VPCs.
+  string dashboard_link = 2;
```

Comment:
This field feels weird; how will it change across different pages?

Reply:
Actually, now that I look at it we also have the link as part of `EC2InstanceConnectEndpoint`. If this is redundant, we may be able to remove the field from `ListEICEResponse`?

```proto
// DashboardLink is a URL to AWS Console where the user can see the EC2 Instance Connect Endpoint.
  string dashboard_link = 4;
```

Reply:
> Actually, now that I look at it we also have the link as part of `EC2InstanceConnectEndpoint`. If this is redundant, we may be able to remove the field from `ListEICEResponse`?
> 
> ```proto
> // DashboardLink is a URL to AWS Console where the user can see the EC2 Instance Connect Endpoint.
>   string dashboard_link = 4;
> ```

That one is specific to a single VPC Endpoint, and not for the AWS page where it lists all the EICE endpoints.

Reply:
Fair enough!